src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -1494,38 +1488,40 @@ internal Type[] FunctionPointerReturnAndParameterTypes
                 }
             }
 
-            internal string? GetName(TypeNameKind kind)
+            // No namespace, full instantiation, and assembly.
+            internal string GetName() => ConstructName(ref m_name, TypeNameFormatFlags.FormatBasic)!;
```

Comment:
```suggestion
            internal string GetName() => m_name ?? ConstructName(ref m_name, TypeNameFormatFlags.FormatBasic)!;
```
Nit: Should we inline the fast path here to make the pattern more uniform?